File: test-do/src/app/shared/models/todoList.ts
import { ToDoItem } from './todoItem';
import { ToDoProject } from './todoProject';

export class ToDoList {
    constructor(
        public name: string = '',
        public owner: string = '',
        public projectId: number,
        public id?: number,
        public items?: ToDoItem[],
        public project?: ToDoProject) {}
}

File: TestDoServer/DAL/Models/ToDoList.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;
using TestDoServer.Models;

namespace TestDoServer.DAL.Models
{
    public class ToDoList
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Owner { get; set; }
        [ForeignKey("Project")]
        public int ProjectId { get; set; }

        public ToDoProject Project { get; set; }
        public List<ToDoItem> Items { get; set; }
    }
}
